### PR TITLE
NetworkStatusDataSource

### PR DESCRIPTION
**Pull request by nelsitoPuglisi**
## Goal

The goal is to capture network connectivity changes as network status spans, instead of network status breadcrumb

## Testing

Unit, integration and manual

### CODE REVIEW

**Comment by nelsitoPuglisi** [comment]
* **#775** <a href="https://app.graphite.dev/github/pr/embrace-io/embrace-android-sdk/775?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @nelsitoPuglisi and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: embrace-android-sdk/src/test/java/io/embrace/android/embracesdk/EmbraceNetworkConnectivityServiceTest.kt [hunk 1]
```diff
@@ -96,85 +95,6 @@ internal class EmbraceNetworkConnectivityServiceTest {
         verify { context.unregisterReceiver(service) }
     }
 
-    @Test
```
Comment (bidetofevil) [inline embrace-android-sdk/src/test/java/io/embrace/android/embracesdk/EmbraceNetworkConnectivityServiceTest.kt:99]: Do we have equivalent tests for the data source or are these functionality implemented by the infra the data source builds on?

**Review by bidetofevil — APPROVED** [review]
LGTM. One question about the unit tests - and if there's anything @fractalwrench wants to explicitly look at, feel free to wait for him. But looks good to me

**Review by fractalwrench — APPROVED** [review]
LGTM

File: embrace-android-sdk/src/test/java/io/embrace/android/embracesdk/EmbraceNetworkConnectivityServiceTest.kt (hunk 1 above)
Comment (nelsitoPuglisi) [reply embrace-android-sdk/src/test/java/io/embrace/android/embracesdk/EmbraceNetworkConnectivityServiceTest.kt:99]: So, the new data source doesn't care about the actual network status or connection type. I can write a new set of unit tests to verify that each type is returned properly by `EmbraceNetworkConnectivityService` but I don't see much value there TBH.

**Pull request merged**